### PR TITLE
increase GNRC_NETIF_IPV6_ADDRS_NUMOF

### PR DESCRIPTION
**Pull request by MrKevinWeiss**
### Contribution description

It seems something happened and we have no available slot for adding an address.  This makes the release tests fail.

Here is a solution but maybe someone has something better

### Testing procedure

```
BUILD_IN_DOCKER=1 BOARD=samr21-xpro make flash term -C tests/net/gnrc_udp/
```

```
ifconfig 5 add beef::1
```

It should be able to add.

### CODE REVIEW

**Comment by chrysn** [comment]
Should we just increase the default? The default values should be usable, and those who need a memory minimum can still tweak it down.

That way it'd also solve an issue in https://github.com/RIOT-OS/RIOT/pull/16840 which otherwise needs a not-quite-kosher filtering of which prefix is the better one at the 6LBR.

File: examples/gnrc_networking/Makefile [hunk 1]
```diff
@@ -60,6 +60,10 @@ endif
 # DODAG Configuration Options (see the doc for more info)
 # CFLAGS += -DCONFIG_GNRC_RPL_DODAG_CONF_OPTIONAL_ON_JOIN
 
+ifndef CONFIG_GNRC_NETIF_IPV6_ADDRS_NUMOF
+  CFLAGS += -DCONFIG_GNRC_NETIF_IPV6_ADDRS_NUMOF=2
```
Comment (benpicco) [inline examples/gnrc_networking/Makefile:64]: But 2 is already the default

Comment (MrKevinWeiss) [reply examples/gnrc_networking/Makefile:64]: Doh, I was just confirming it was failing there...  Thanks!

**Comment by MrKevinWeiss** [comment]
> Should we just increase the default? The default values should be usable, and those who need a memory minimum can still tweak it down.
> 
> That way it'd also solve an issue in #16840 which otherwise needs a not-quite-kosher filtering of which prefix is the better one at the 6LBR.

I think the philosophy of RIOT is more about keeping it small by default, I would rather get an error saying I need more space (maybe the error message can be improved).  It is easier to get to the perfect solution™ that way than silently reserving data that is not being used.

If anyone feels strongly in either direction I am happy to support it.  I just want the examples/tests to pass the release specs!

**Comment by benpicco** [comment]
Is that 'something happened' that you now have a global address in addition to the link-local one?

Can you share the output of `ifconfig`?

**Comment by chrysn** [comment]
I'd say it's about keeping *reasonably* small defaults, and provided you can get the message through.

If every tutorial that's supposed to run on a user's network starts with "Because we don't know your network config, let's be safe and set GNRC_NETIF_IPV6_ADDRS_NUMOF to 4" (or "let's set up an isolated network so we're sure things work"), I think that's where practicality and usability overrules compact defaults. More so if also our own tests have to overwrite it.

**Comment by benpicco** [comment]
Reducing memory footprint is always much harder than increasing it because you have 1000 little cuts to tweak.

I think the default of two addresses is sensible, if you need more you might as well bump that config value.

**Comment by MrKevinWeiss** [comment]
> Is that 'something happened' that you now have a global address in addition to the link-local one?

Yup, I just didn't have the time to find out why it was added.

> Can you share the output of ifconfig?

The [release output](https://github.com/RIOT-OS/RIOT/actions/runs/7630819374/job/20787346890#step:15:305) has it.  I also was able to reproduce it on the `samr21-xpro` locally (but I may not be able to go into the office today to get the output again).
Here is a failure for the [gnrc_networking example](https://github.com/RIOT-OS/RIOT/actions/runs/7630819374/job/20787346890#step:15:1952)

> For me this works just fine on examples/gnrc_networking without having to increase the number of addresses.

It seems like you don't have the default global address in your `ifconfig` compared to the ones in the RC1 action run.

**Comment by benpicco** [comment]
> It seems like you don't have the default global address in your `ifconfig` compared to the ones in the RC1 action run.

There is no such thing as a 'default global address' - looks like you have a border router running in your setup that announces that address.


You can check `nib route` or `nib abr`.

**Comment by MrKevinWeiss** [comment]
Thanks @benpicco for clarifying, it would seem like there is a border router not only somewhere in IOT labs when I was trying to run the tests (and the flash actually worked) but also in locally.

> try setting a different PAN ID.

I think that the best way forward would be something like this, if there is already a global address then change the PAN ID or channel and remote the assigned one and replace it with the known.  Since this would be a release specs change I think we can close this.

**Pull request closed without merging**